ffmpeg/ffmpeg.go

```diff
@@ -120,15 +121,26 @@ func accelDeviceType(accel Acceleration) (C.enum_AVHWDeviceType, error) {
 
 func Transcode2(input *TranscodeOptionsIn, ps []TranscodeOptions) error {
 
+	_, ret := Transcode3(input, ps)
+	return ret
+}
+
+type TranscodeReceipt struct {
+	Frames int64
+	Pixels int64
+}
+
+func Transcode3(input *TranscodeOptionsIn, ps []TranscodeOptions) (res TranscodeReceipt, err error) {
```

Comment:
Should we return `res` as a pointer rather than a value here?

Reply:
We can, but this is golang and we have garbage collection.   Can you expand on your thoughts here?  Why do you want to return by pointer rather than by reference?   It's bad practice to do this in golang unless you have a good reason for it.

Reply:
> It's bad practice to do this in golang unless you have a good reason for it.

Can you point me to guidance describing this practice? If anything, returning a pointer is leaning more on the GC.

Does the golang compiler implicitly convert `res` to a heap allocated reference? My understanding is that it doesn't, and we would be making full copies of the struct each time it's passed or returned, which seems undesirable.

Also, returning a nullable pointer seems a little more explicit to indicate nonexistent data, rather than zero values which could inadvertently be interpreted as valid data.